turbinia/jobs/analysis/docker_logs.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Job to execute Docker analysis task."""
+from __future__ import unicode_literals
+
+from turbinia.workers import artifact
+
+from turbinia.evidence import Directory
+from turbinia.evidence import RawDisk
+from turbinia.evidence import GoogleCloudDisk
+from turbinia.evidence import GoogleCloudDiskRawEmbedded
+from turbinia.evidence import ExportedFileArtifact
+from turbinia.evidence import ReportText
+
+from turbinia.jobs import TurbiniaJob
+from turbinia.workers.analysis import wordpress
+
+class DockerLogExtractionJob(TurbiniaJob):
+  """Docker log extraction job."""
+
+  evidence_input = [
+      Directory, RawDisk, GoogleCloudDisk, GoogleCloudDiskRawEmbedded]
+
+  evidence_output = [ExportedFileArtifact]
+
+  def __init__(self):
+    super(DockerLogExtractionJob, self).__init__(
+        name='DockerLogExtractionJob')
+
+  def create_tasks(self, evidence):
+    """Create task.
+
+    Args:
+      evidence: List of evidence object to process
+
+    Returns:
+        A list of tasks to schedule.
+    """
+    tasks = [artifact.FileArtifactExtractionTask('DockerContainerLogs') for _
+             in evidence]
+    return tasks
+
+class DockerLogAnalysisJob(TurbiniaJob):
```

Comment:
Actually, I missed one big thing here.  Why would these Tasks only be specific to Docker containers[1].  Can we at minimum add separate Tasks for the local system logs?  I'm assuming that shouldn't require any changes to the Tasks themselves, and is maybe just another Artifact that needs to be referenced?

We don't have to do it in this PR, but I wonder if there is a better way to handle this architecturally as well, for example possibly adding a Docker enumeration Task that creates a new DockerContainer evidence type.  That might not work without some extra effort though, because the two evidence types may be too fundamentally different to be handled generically in a Task.  I wonder if there is a reliable way to do a mount of the container image to make it more generic for the Tasks?  cc: @berggren and @rgayon for thoughts on this part.

[1]  This brings up another big point that we should make our other analysis Tasks should also be looking to 'recurse' into Docker containers as well

Reply:
Yes, this came up during a chat with @rgayon earlier this week.

In theory we could use the exact same plugin with a minor change in the artifact. The "docker log" format is slightly different than a normal nginx or apache access log (it's, essentially, access logs wrapped in JSON). The analysis logic ignores the JSON and just greps through the output, which will work on both JSON-wrapped logs and regular access logs.

I'm happy to rename `DockerLogExtractionJob` to `AccessLogExtractionJob` to reflect regular log grabbing using different artifacts (nginx, apache, and also docker) and passing that on to the existing Task. Can we just add an extra `FileArtifactExtractionTask` in [L51](https://github.com/google/turbinia/pull/240/files#diff-009fc58bd66af90368a40aa9445ad0deR51).

Finally, there's a bit of a nuance in Docker logs - the cases I was thinking to tackle with this Job was when the container logs to `stdout` and Docker logs that in its log. In theses cases, if you mount the docker container as a regular FS, you won't find logs anywhere in the mountpoint since they are all sent outside of the container.

Reply:
Yeah, that would be great if we can rename it and just add an extra Artifact extraction.  As you suggest, just adding it to L51 should work if there is already a suitable artifact type.